application/heartbeater.go

```diff
@@ -62,6 +65,15 @@ func (hb *heartbeater) beat() {
 		tags["component"] = component
 		hb.send(tags)
 	}
+
+	//send customTags heartbeat
+	hb.mux.Lock()
+	for len(hb.customTags) > 0 {
+		tags := hb.customTags[0]
+		hb.customTags = hb.customTags[1:]
```

Comment:
This mutates the internal customTags. It will be empty the next time `beat()` runs.

Reply:
it should be empty